Mathlib/GroupTheory/GroupAction/Hom.lean

```diff
@@ -1,212 +1,524 @@
-Authors: Kenny Lau
+Authors: Kenny Lau, Antoine Chambert-Loir
+
+! This file was ported from Lean 3 source module algebra.hom.group_action
+! leanprover-community/mathlib commit e7bab9a85e92cf46c02cb4725a7be2f04691e3a7
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
 -/
+
 import Mathlib.Algebra.GroupRingAction.Basic
 import Mathlib.Algebra.Module.Basic
-
-#align_import algebra.hom.group_action from "leanprover-community/mathlib"@"e7bab9a85e92cf46c02cb4725a7be2f04691e3a7"
+import Mathlib.Algebra.Ring.Equiv
 
 /-!
 # Equivariant homomorphisms
 
 ## Main definitions
 
-* `MulActionHom M X Y`, the type of equivariant functions from `X` to `Y`, where `M` is a monoid
-  that acts on the types `X` and `Y`.
-* `DistribMulActionHom M A B`, the type of equivariant additive monoid homomorphisms
-  from `A` to `B`, where `M` is a monoid that acts on the additive monoids `A` and `B`.
-* `MulSemiringActionHom M R S`, the type of equivariant ring homomorphisms
-  from `R` to `S`, where `M` is a monoid that acts on the rings `R` and `S`.
+* `MulActionHom φ X Y`, the type of equivariant functions from `X` to `Y`,
+  where `φ : M → N` is a map, `M` acting on the type `X` and `N` acting on the type of `Y`.
+* `DistribMulActionHom φ A B`,
+  the type of equivariant additive monoid homomorphisms from `A` to `B`,
+  where `φ : M → N` is a morphism of monoids,
+  `M` acting on the additive monoid `A` and `N` acting on the additive monoid of `B`
+* `SMulSemiringHom φ R S`, the type of equivariant ring homomorphisms
+  from `R` to `S`, where `φ : M → N` is a morphism of monoids,
+  `M` acting on the ring `R` and `N` acting on the ring `S`.
 
 The above types have corresponding classes:
-* `SMulHomClass F M X Y` states that `F` is a type of bundled `X → Y` homs
-  preserving scalar multiplication by `M`
-* `DistribMulActionHomClass F M A B` states that `F` is a type of bundled `A → B` homs
-  preserving the additive monoid structure and scalar multiplication by `M`
-* `MulSemiringActionHomClass F M R S` states that `F` is a type of bundled `R → S` homs
-  preserving the ring structure and scalar multiplication by `M`
+* `MulActionHomClass F φ X Y` states that `F` is a type of bundled `X → Y` homs
+  which are `φ`-equivariant
+* `DistribMulActionHomClass F φ A B` states that `F` is a type of bundled `A → B` homs
+  preserving the additive monoid structure and `φ`-equivariant
+* `SMulSemiringHomClass F φ R S` states that `F` is a type of bundled `R → S` homs
+  preserving the ring structure and `φ`-equivariant
 
 ## Notations
 
-* `X →[M] Y` is `MulActionHom M X Y`.
-* `A →+[M] B` is `DistribMulActionHom M A B`.
-* `R →+*[M] S` is `MulSemiringActionHom M R S`.
+* `X →ₑ[φ] Y` is `MulActionHom φ X Y`.
+* `A →ₑ+[φ] B` is `DistribMulActionHom φ A B`.
+* `R →ₑ+*[φ] S` is `MulSemiringActionHom φ R S`.
+
+* When `M = N` and `φ = MonoidHom.id M`, we provide the notation `X →[M] Y`
+-/
+
+section CompTriple
```

Comment:
Like `RingHomCompTriple`, I think this section deserves its own file. Since it's about pure functions, maybe `Logic/Function/CompTypeclasses.lean`?

Reply:
It uses a bit of groups, `Algebra/Group/Home/CompTypeClass`.

Reply:
Logic/Function/CompTypeclass
(plural?)

Reply:
Either singular or plural works for me!